web/src/modules/i18n/BrowserLanguageDetector.js

```diff
@@ -0,0 +1,65 @@
+let hasLocalStorageSupport = null
+
+const localStorageAvailable = () => {
+  if (hasLocalStorageSupport !== null) {
+    return hasLocalStorageSupport
+  }
+
+  try {
+    hasLocalStorageSupport = window !== 'undefined' && window.localStorage !== null
+    const testKey = 'i18next.translate.boo'
+    window.localStorage.setItem(testKey, 'foo')
+    window.localStorage.removeItem(testKey)
+  } catch (e) {
+    hasLocalStorageSupport = false
+  }
+  return hasLocalStorageSupport
+}
+
+const LANGUAGE_LOCAL_STORAGE = 'i18nextLng'
+
+export default {
+  type: 'languageDetector',
+  async: false,
+  init: (services, detectorOptions, i18nextOptions) => {},
+  // https://github.com/i18next/i18next-browser-languageDetector/blob/master/src/browserLookups/navigator.js
+  // Returns array of ISO-639-2 or ISO-639-3 language codes
+  detect: () => {
+    const bcp47Tags = []
+
+    if (localStorageAvailable()) {
+      const localStorageLanguage = window.localStorage.getItem(LANGUAGE_LOCAL_STORAGE)
+      if (localStorageLanguage) {
+        bcp47Tags.push(localStorageLanguage)
+      }
+    }
+
+    if (typeof navigator !== 'undefined') {
+      if (navigator.languages) { // chrome only; not an array, so can't use .push.apply instead of iterating
+        for (let i = 0; i < navigator.languages.length; i++) {
+          bcp47Tags.push(navigator.languages[i])
+        }
+      }
+      if (navigator.userLanguage) {
+        bcp47Tags.push(navigator.userLanguage)
+      }
+      if (navigator.language) {
+        bcp47Tags.push(navigator.language)
+      }
+    }
+
+    const iso639Codes = bcp47Tags.map(bcp47Tag => {
```

Comment:
We get BCP47 codes from the browser. This is a simple way of extracting the primary language code.

We need to do this for now as we can not match en-US to the backend language en

Reply:
We need to decide whether this is oke we whether we implement code to skip the "Language not available" dialog by automatically matching en-US to en. We can use the languages endpoint to implement a heuristic for choosing a language.

Reply:
I think there should not be a `Language not available` dialog for `en-US`. I think this is okay for now.